src/lex.cpp

```diff
@@ -1288,6 +1292,58 @@ void lexer::parse_modern_octal_number() {
       diag_unexpected_characters_in_octal_number>(input);
 }
 
+void lexer::check_integer_precision_loss(string8_view number_literal) {
+  // Any integer which is 15 or fewer digits is guaranteed to be able to be
+  // represented accurately without precision loss.
```

Comment:
This doesn't explain where 15 comes from. Is it based on 1<<53 or something? (Then I'd guess you'd need to explain where 53 comes from.)